storage/dynamodb/dynamodb.go

```diff
@@ -0,0 +1,742 @@
+		},
+		TableName: aws.String(c.table)},
+	)
+
+	if err != nil {
+		c.logger.Infof("%v\n", err)
+		return keys, fmt.Errorf("%v", err)
+	}
+
+	var keyResp Keys
+	err = attributevalue.UnmarshalMap(resp.Item, &keyResp)
+
+	storageKey := toStorageKey(keyResp)
+
+	if err != nil {
+		c.logger.Infof("%v\n", err)
+	}
+
+	return storageKey, err
+}
+
+func (c *conn) GetRefresh(id string) (storage.RefreshToken, error) {
+	var temp storage.RefreshToken
+	resp, err := c.getItem(refreshTokenKey, id)
+
+	if err != nil {
+		return temp, err
+	}
+
+	var refreshDbd RefreshToken
+	err = attributevalue.UnmarshalMap(resp.Item, &refreshDbd)
+
+	if err != nil {
+		return temp, err
+	}
+
+	temp = refreshDbd.Token
+	return temp, nil
+}
+
+func (c *conn) GetPassword(email string) (storage.Password, error) {
+	var temp storage.Password
+	resp, err := c.getItem(passwordKey, strings.ToLower(email))
+
+	if err != nil {
+		return temp, err
+	}
+
+	var passwordDbd Password
+	err = attributevalue.UnmarshalMap(resp.Item, &passwordDbd)
+
+	if err != nil {
+		return temp, err
+	}
+
+	temp = passwordDbd.Password
+	return temp, nil
+}
+
+func (c *conn) GetOfflineSessions(userID string, connID string) (storage.OfflineSessions, error) {
+	var temp storage.OfflineSessions
+	resp, err := c.getItem(offlineSessionKey, userID+"-"+connID)
+
+	if err != nil {
+		return temp, err
+	}
+
+	var offlineSessionDbd OfflineSessions
+	err = attributevalue.UnmarshalMap(resp.Item, &offlineSessionDbd)
+
+	if err != nil {
+		return temp, err
+	}
+
+	temp = offlineSessionDbd.Session
+	return temp, nil
+}
+
+func (c *conn) GetConnector(id string) (storage.Connector, error) {
+	var temp storage.Connector
+	resp, err := c.getItem(connectorKey, id)
+
+	if err != nil {
+		return temp, err
+	}
+
+	var connectorDbd Connector
+	err = attributevalue.UnmarshalMap(resp.Item, &connectorDbd)
+
+	if err != nil {
+		return temp, err
+	}
+
+	temp = connectorDbd.Connector
+	return temp, nil
+}
+
+func (c *conn) GetDeviceRequest(userCode string) (storage.DeviceRequest, error) {
+	return storage.DeviceRequest{
+		UserCode:     "",
+		DeviceCode:   "",
+		ClientID:     "",
+		ClientSecret: "",
+		Scopes:       nil,
+		Expiry:       time.Time{},
+	}, nil
+}
+
+func (c *conn) GetDeviceToken(deviceCode string) (storage.DeviceToken, error) {
+	return storage.DeviceToken{
+		DeviceCode:          "",
+		Status:              "",
+		Token:               "",
+		Expiry:              time.Time{},
+		LastRequestTime:     time.Time{},
+		PollIntervalSeconds: 0,
+		PKCE:                storage.PKCE{},
+	}, nil
+}
+
+func (c *conn) getItemsWithKeyPrefix(content_type string) ([]map[string]types.AttributeValue, error) {
+	keyEx := expression.Key("ContentType").Equal(expression.Value(content_type))
+	expr, _ := expression.NewBuilder().WithKeyCondition(keyEx).Build()
+	queryInput := dynamodb.QueryInput{
+		TableName:                 aws.String(c.table),
+		IndexName:                 aws.String("ContentType-index"),
```

Comment:
How is this index defined? Is it projecting only the keys, only some attributes(using INCLUDE) or everything(using ALL) ?

If we are projecting all the attributes and the ContentType is the PK, I think it won't be necessary to use this index.